src/xdp/rx.c

```diff
@@ -1082,6 +1095,12 @@ XdpRxQueueSetProgram(
         SwapParams.RxQueue = RxQueue;
         SwapParams.NewProgram = Program;
         XdpRxQueueSync(RxQueue, XdpRxQueueSwapProgram, &SwapParams);
+        if (OldCompiledProgram != NULL) {
+            //
+            // Free the old compiled program because it's no longer being used.
+            //
+            ExFreePoolWithTag(OldCompiledProgram, XDP_POOLTAG_PROGRAM);
```

Comment:
IMO we should have the allocs and frees both within program.c.

Reply:
I wonder what would be a cleaner way to do this? return the old program pointer on a `_Out_` parameter or pass in a clean up callback?

Reply:
Within this model, we have (or had) the caller (in program.c) retrieving the old program and freeing it upon successfully setting the new program. Can that still work?

Reply:
That can work if we expose RxQueue->Program to program.c again.

Reply:
Updated